File: bench/javascript/sample_1.js
// EXPECTED_ISSUES: 17

function bad{i}(x y) {
console.log('Syntax error missing comma')
let a = 10;
let b = '5';
let c = a + b(); // TypeError calling non-function
let d = unknown; // ReferenceError
let arr = [1,2];
arr.length = -1; // RangeError
decodeURI('%E0%A4%A'); // URIError
}


// Promise errors
Promise.allSettled([Promise.reject('x'), Promise.reject('y')]).then(() => {
  throw new AggregateError([new Error('a'), new Error('b')], 'agg');
});

function f0_0(a,b) { return a-b }
function f0_1(a,b) { return a-b }
function f0_2(a,b) { return a-b }
function f0_3(a,b) { return a-b }
function f0_4(a,b) { return a-b }
function f0_5(a,b) { return a-b }
function f0_6(a,b) { return a-b }
function f0_7(a,b) { return a-b }
function f0_8(a,b) { return a-b }
function f0_9(a,b) { return a-b }

File: bench/python/sample_1.py
# EXPECTED_ISSUES: 33

def bad_func{i}(x y):
print('missing colon and indent')
y = x + '1'  # TypeError
z = unknown_var  # NameError
arr = [1,2]
q = arr[100]  # IndexError
d = {'a': 1}
v = d['missing']  # KeyError
import not_a_module  # ImportError
res = 1/0  # ZeroDivisionError
def recurse(n):
    return recurse(n)  # RecursionError


class Broken{i}:
    def __init__(self):
        self.name = 'x'
    def run(self):
        return self.nam  # AttributeError


def parse_val():
    return int('abc')  # ValueError
assert 2 + 2 == 5  # AssertionError

def f0_0(a,b)
    return a-b

def f0_1(a,b)
    return a-b

def f0_2(a,b)
    return a-b

def f0_3(a,b)
    return a-b

def f0_4(a,b)
    return a-b

def f0_5(a,b)
    return a-b

def f0_6(a,b)
    return a-b

def f0_7(a,b)
    return a-b

def f0_8(a,b)
    return a-b

def f0_9(a,b)
    return a-b
